### PR TITLE
Compare screenshots with main on PRs

### PR DESCRIPTION
**Pull request by mockersf**
# Objective

- Compare screenshots for a few examples between PRs and main

## Solution

- Send screenshots taken to a screenshot comparison service
- Not completely sure every thing will work at once, but it shouldn't break anything at least
- it needs a secret to work, I'll add it if enough people agree with this PR
- this PR doesn't change anything on the screenshot selection (load_gltf and breakout currently), this will need rendering folks input and can happen later

### CODE REVIEW

**Comment by hymm** [comment]
What happens when there's a failure? does ci just fail or are you able to see the diffs somewhere?

**Comment by mockersf** [comment]
> What happens when there's a failure? does ci just fail or are you able to see the diffs somewhere?

it should log an url like https://pixel-eagle.vleue.com/project/b25a040a-a980-4602-b90c-d480ab84076d/run/1335/compare/1327 that will let you see the issues

**Comment by mockersf** [comment]
job is currently failing because the secret is not set

**Review by superdump — APPROVED** [review]
Yes please

**Comment by superdump** [comment]
Maybe we should put the flight helmet into the lighting scene. Or just put together a test scene which has a whole bunch of features enabled all at once to catch differences.

**Comment by mockersf** [comment]
> Maybe we should put the flight helmet into the lighting scene. Or just put together a test scene which has a whole bunch of features enabled all at once to catch differences.

Since https://github.com/bevyengine/bevy/pull/11904, the flight helmet makes rendering crash in CI on Windows

File: .github/workflows/send-screenshots-to-pixeleagle.yml
Comment (BD103) [inline .github/workflows/send-screenshots-to-pixeleagle.yml:33]: Can this be set using the `env` property? I feel like it signals better that this is a hardcoded constant, and may be changed in the future.

```yaml
- name: Send to Pixel Eagle
  env:
    project: B04F67C0-C054-4A6F-92EC-F599FEC2FD1D
  run: ...
```

File: .github/workflows/send-screenshots-to-pixeleagle.yml
Comment (BD103) [inline .github/workflows/send-screenshots-to-pixeleagle.yml:22]: This can be changed back to `ubuntu-latest` once [Ubuntu 24 becomes the new default](https://github.blog/changelog/2024-05-14-github-hosted-runners-public-beta-of-ubuntu-24-04-is-now-available/). ([Installed software](https://github.com/actions/runner-images/blob/main/images/ubuntu/Ubuntu2404-Readme.md#installed-apt-packages))

I won't block on it, though, because who knows when that will be ready. :)

File: .github/workflows/validation-jobs.yml
Comment (BD103) [inline .github/workflows/validation-jobs.yml:71]: Could you add a comment on why this is different, since all of the other jobs in this file uses `== 'merge_group'`?

File: .github/workflows/validation-jobs.yml
Comment (BD103) [inline .github/workflows/validation-jobs.yml:140]: Same thing as above, please!

**Review by BD103 — APPROVED** [review]
This will be really nice for rendering PRs, since we can catch regressions before they merge. (Such as the MacOS thing that happened a few days ago.)

I have a few nits, but they're non-blocking. :)

**Comment by alice-i-cecile** [comment]
@mockersf I'm happy to merge this: looks like it needs you to add a secret first?

**Comment by mockersf** [comment]
this *should* be ready to be merged, but can't be unless we remove the job `run-examples-macos-metal` from the required checks: I needed to move it to a `pull_request_target` trigger for it to be able to access secrets, which means it runs from its definition on the base branch of the PR, and this job in this workflow doesn't exist yet there

**Comment by alice-i-cecile** [comment]
I'm on board with that plan: please feel free to do that and merge.

**Comment by mockersf** [comment]
remove the required check, merged the pr, added the required check back, notice the screenshot upload failed 😭 
https://github.com/bevyengine/bevy/pull/15894 should fix it

**Pull request merged**
